agent_handlers_test.go

```diff
@@ -39,14 +39,10 @@ func TestOnConnectionStateChangeCallback(t *testing.T) {
 	}
 
 	aAgent, err := NewAgent(cfg)
-	if err != nil {
-		t.Error(err)
-	}
+	assert.NoError(err)
```

Comment:
I think we shall not use assert except we really need it.
Require is better because test stops when error happens.
With assert test continues, and most likely we don't want that.

Reply:
I see, you already replaced to "require".

Reply:
Yes, I started by refactoring it while maintaining its original behaviour (`t.Error()` => `assert.XXX`).
But you are correct.. I think we often should really use `require` instead.